lib/helpers/promiseOrCallback.js

```diff
@@ -35,7 +34,7 @@ module.exports = function promiseOrCallback(callback, fn, ee, Promise) {
     }
   }
 
-  Promise = Promise || PromiseProvider.get();
+  Promise = Promise || global.Promise || require('bluebird');
```

Comment:
Please get rid of `require('bluebird')` here. In general, just get rid of the `Promise` param.

There's no need to import bluebird here because Mongoose doesn't support any JavaScript runtimes that don't have native promises.